flang/lib/Lower/OpenMP.cpp

```diff
@@ -29,35 +29,48 @@ getDesignatorNameIfDataRef(const Fortran::parser::Designator &designator) {
   return dataRef ? std::get_if<Fortran::parser::Name>(&dataRef->u) : nullptr;
 }
 
+template <typename T>
+static void createPrivateVarSyms(Fortran::lower::AbstractConverter &converter,
+                                 const T *clause) {
+  Fortran::semantics::Symbol *sym = nullptr;
+  const Fortran::parser::OmpObjectList &ompObjectList = clause->v;
+  for (const auto &ompObject : ompObjectList.v) {
+    std::visit(
+        Fortran::common::visitors{
+            [&](const Fortran::parser::Designator &designator) {
+              if (const auto *name = getDesignatorNameIfDataRef(designator)) {
+                sym = name->symbol;
+              }
+            },
+            [&](const Fortran::parser::Name &name) { sym = name.symbol; }},
```

Comment:
What kind of variables is this `Name` used for? As I tested so far, common block is parsered as `Name`. Can common block be in `firstprivate` clause?